internal/output/html.go

```diff
@@ -335,11 +381,89 @@ func buildHTMLResult(ecosystemMap map[string][]HTMLSourceResult, resultCount HTM
 	})
 
 	ecosystemResults = append(ecosystemResults, osResults...)
+	var layers []LayerInfo
+	if isContainerScanning {
+		layers = getAllLayers(ecosystemResults)
+	}
+	vulnTypeCount := getVulnTypeCount(ecosystemResults)
 
 	return HTMLResult{
-		EcosystemResults: ecosystemResults,
-		HTMLVulnCount:    resultCount,
+		EcosystemResults:    ecosystemResults,
+		HTMLVulnCount:       resultCount,
+		IsContainerScanning: isContainerScanning,
+		AllLayers:           layers,
+		VulnTypeCount:       vulnTypeCount,
+	}
+}
+
+func getVulnTypeCount(result []HTMLEcosystemResult) VulnTypeCount {
+	osType := 0
+	projectType := 0
+	uncalled := 0
+	all := 0
+
+	for _, ecosystem := range result {
+		for _, source := range ecosystem.Sources {
+			if ecosystem.IsOS {
+				osType += source.HTMLVulnCount.Called
+			} else {
+				projectType += source.HTMLVulnCount.Called
+			}
+			all += source.HTMLVulnCount.Called
+			uncalled += source.HTMLVulnCount.Uncalled
+		}
+	}
+
+	return VulnTypeCount{
+		All:      all,
+		OS:       osType,
+		Project:  projectType,
+		Uncalled: uncalled,
+	}
+}
+
+func getAllLayers(result []HTMLEcosystemResult) []LayerInfo {
+	layerMap := make(map[string]string)
+	layerCount := make(map[string]HTMLVulnCount)
+	layerIndex := 0
+
+	for _, ecosystem := range result {
+		for _, source := range ecosystem.Sources {
+			for _, packageInfo := range source.PackageResults {
+				layerID := packageInfo.HTMLPackageDetail.LayerID
+				layerCommand := packageInfo.HTMLPackageDetail.LayerCommand
+
+				// Check if this layer ID and command combination is already in the map
+				if _, ok := layerMap[layerID]; !ok {
+					var resultCount HTMLVulnCount
+					updateCount(&resultCount, &packageInfo.HTMLVulnCount)
```

Comment:
nit: probably worth exploring at some point if it would be cleaner to make this a method on `HTMLVulnCount`

(pointers always confuse me since I don't work with them day-to-day, so I won't say off the top of my head that it would definitely be worth it 😅)

Reply:
After this PR is merged, I'll move the HTMLResult struct from html.go to table.go to create a package summary for table output during container scanning. I plan to refactor the code then and will try to make this function cleaner. 

(I also don't like pointers.)